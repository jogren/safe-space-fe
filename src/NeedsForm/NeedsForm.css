.items-container {
  overflow-y: scroll;
}

#necessities-header {
  text-align: center;
  font-size: 30px;
  margin: 15px;
}