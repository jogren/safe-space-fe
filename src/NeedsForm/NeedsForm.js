import React, { Component } from 'react'

class NeedsForm extends Component {
    // constructor() {
    //     super(); 
    //     this.state = {
    //         checked: false 
    //     }
    // }

    // toggleChecked = (e) => {
    //     e.preventDefault(); 
    //     this.setState({ checked: !this.state.checked })
    // }

    render() {
        // const allItems = this.props.items.map(item => {
        //     return <article className="item">
        //         <button id="checkbox-button" type="checkbox" checked={this.state.checked} onClick={this.toggleChecked}></button>
        //         <p>{item}</p>
        //         </article>
        // })

        return (
            <section>
                {/* {allItems} */}
                <h1>Hi</h1>
            </section>
        )
    }
}

export default NeedsForm; 