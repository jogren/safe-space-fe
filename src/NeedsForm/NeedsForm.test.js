import React from "react";
import { mount } from "enzyme";
import NeedsForm from "./NeedsForm";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

describe("NeedsForm", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(
        <NeedsForm
        items={[
        "Diapers",
        "Baby Wipes",
        "Breastfeeding Supplies",
        "Infant Formula",
        "Feminine Products",
        "Phone Charger (iphone)",
        "Phone Charger (android)",
        "add item"
        ]}
        setItems={jest.fn()}
        neededItems={[]}
        setNeededItems={jest.fn()}
      />
    );
  });

  it("should render without crashing", () => {
    expect(wrapper.length).toEqual(1);
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});