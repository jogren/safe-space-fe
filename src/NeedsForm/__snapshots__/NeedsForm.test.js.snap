// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NeedsForm should match the snapshot 1`] = `
<NeedsForm
  items={
    Array [
      "Diapers",
      "Baby Wipes",
      "Breastfeeding Supplies",
      "Infant Formula",
      "Feminine Products",
      "Phone Charger (iphone)",
      "Phone Charger (android)",
      "add item",
    ]
  }
  neededItems={Array []}
  setItems={[MockFunction]}
  setNeededItems={[MockFunction]}
>
  <section
    className="NeedsForm"
  >
    <h1
      id="necessities-header"
    >
      Necessities:
    </h1>
    <article
      id="items-container"
    >
      <Item
        item="Diapers"
        items={
          Array [
            "Diapers",
            "Baby Wipes",
            "Breastfeeding Supplies",
            "Infant Formula",
            "Feminine Products",
            "Phone Charger (iphone)",
            "Phone Charger (android)",
            "add item",
          ]
        }
        key="0"
        neededItems={Array []}
        setItems={[MockFunction]}
        setNeededItems={[MockFunction]}
      >
        <article
          className="Item"
        >
          <div
            id="checkbox-item-container"
          >
            <img
              alt=""
              id="check-or-uncheck"
              onClick={[Function]}
              src={
                Object {
                  "ReactComponent": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "render": [Function],
                  },
                  "__esModule": true,
                  "default": "unchecked.svg",
                }
              }
            />
            <h3
              id="item-name"
              onClick={[Function]}
            >
              Diapers
            </h3>
          </div>
        </article>
      </Item>
      <Item
        item="Baby Wipes"
        items={
          Array [
            "Diapers",
            "Baby Wipes",
            "Breastfeeding Supplies",
            "Infant Formula",
            "Feminine Products",
            "Phone Charger (iphone)",
            "Phone Charger (android)",
            "add item",
          ]
        }
        key="1"
        neededItems={Array []}
        setItems={[MockFunction]}
        setNeededItems={[MockFunction]}
      >
        <article
          className="Item"
        >
          <div
            id="checkbox-item-container"
          >
            <img
              alt=""
              id="check-or-uncheck"
              onClick={[Function]}
              src={
                Object {
                  "ReactComponent": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "render": [Function],
                  },
                  "__esModule": true,
                  "default": "unchecked.svg",
                }
              }
            />
            <h3
              id="item-name"
              onClick={[Function]}
            >
              Baby Wipes
            </h3>
          </div>
        </article>
      </Item>
      <Item
        item="Breastfeeding Supplies"
        items={
          Array [
            "Diapers",
            "Baby Wipes",
            "Breastfeeding Supplies",
            "Infant Formula",
            "Feminine Products",
            "Phone Charger (iphone)",
            "Phone Charger (android)",
            "add item",
          ]
        }
        key="2"
        neededItems={Array []}
        setItems={[MockFunction]}
        setNeededItems={[MockFunction]}
      >
        <article
          className="Item"
        >
          <div
            id="checkbox-item-container"
          >
            <img
              alt=""
              id="check-or-uncheck"
              onClick={[Function]}
              src={
                Object {
                  "ReactComponent": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "render": [Function],
                  },
                  "__esModule": true,
                  "default": "unchecked.svg",
                }
              }
            />
            <h3
              id="item-name"
              onClick={[Function]}
            >
              Breastfeeding Supplies
            </h3>
          </div>
        </article>
      </Item>
      <Item
        item="Infant Formula"
        items={
          Array [
            "Diapers",
            "Baby Wipes",
            "Breastfeeding Supplies",
            "Infant Formula",
            "Feminine Products",
            "Phone Charger (iphone)",
            "Phone Charger (android)",
            "add item",
          ]
        }
        key="3"
        neededItems={Array []}
        setItems={[MockFunction]}
        setNeededItems={[MockFunction]}
      >
        <article
          className="Item"
        >
          <div
            id="checkbox-item-container"
          >
            <img
              alt=""
              id="check-or-uncheck"
              onClick={[Function]}
              src={
                Object {
                  "ReactComponent": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "render": [Function],
                  },
                  "__esModule": true,
                  "default": "unchecked.svg",
                }
              }
            />
            <h3
              id="item-name"
              onClick={[Function]}
            >
              Infant Formula
            </h3>
          </div>
        </article>
      </Item>
      <Item
        item="Feminine Products"
        items={
          Array [
            "Diapers",
            "Baby Wipes",
            "Breastfeeding Supplies",
            "Infant Formula",
            "Feminine Products",
            "Phone Charger (iphone)",
            "Phone Charger (android)",
            "add item",
          ]
        }
        key="4"
        neededItems={Array []}
        setItems={[MockFunction]}
        setNeededItems={[MockFunction]}
      >
        <article
          className="Item"
        >
          <div
            id="checkbox-item-container"
          >
            <img
              alt=""
              id="check-or-uncheck"
              onClick={[Function]}
              src={
                Object {
                  "ReactComponent": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "render": [Function],
                  },
                  "__esModule": true,
                  "default": "unchecked.svg",
                }
              }
            />
            <h3
              id="item-name"
              onClick={[Function]}
            >
              Feminine Products
            </h3>
          </div>
        </article>
      </Item>
      <Item
        item="Phone Charger (iphone)"
        items={
          Array [
            "Diapers",
            "Baby Wipes",
            "Breastfeeding Supplies",
            "Infant Formula",
            "Feminine Products",
            "Phone Charger (iphone)",
            "Phone Charger (android)",
            "add item",
          ]
        }
        key="5"
        neededItems={Array []}
        setItems={[MockFunction]}
        setNeededItems={[MockFunction]}
      >
        <article
          className="Item"
        >
          <div
            id="checkbox-item-container"
          >
            <img
              alt=""
              id="check-or-uncheck"
              onClick={[Function]}
              src={
                Object {
                  "ReactComponent": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "render": [Function],
                  },
                  "__esModule": true,
                  "default": "unchecked.svg",
                }
              }
            />
            <h3
              id="item-name"
              onClick={[Function]}
            >
              Phone Charger (iphone)
            </h3>
          </div>
        </article>
      </Item>
      <Item
        item="Phone Charger (android)"
        items={
          Array [
            "Diapers",
            "Baby Wipes",
            "Breastfeeding Supplies",
            "Infant Formula",
            "Feminine Products",
            "Phone Charger (iphone)",
            "Phone Charger (android)",
            "add item",
          ]
        }
        key="6"
        neededItems={Array []}
        setItems={[MockFunction]}
        setNeededItems={[MockFunction]}
      >
        <article
          className="Item"
        >
          <div
            id="checkbox-item-container"
          >
            <img
              alt=""
              id="check-or-uncheck"
              onClick={[Function]}
              src={
                Object {
                  "ReactComponent": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "render": [Function],
                  },
                  "__esModule": true,
                  "default": "unchecked.svg",
                }
              }
            />
            <h3
              id="item-name"
              onClick={[Function]}
            >
              Phone Charger (android)
            </h3>
          </div>
        </article>
      </Item>
      <Item
        item="add item"
        items={
          Array [
            "Diapers",
            "Baby Wipes",
            "Breastfeeding Supplies",
            "Infant Formula",
            "Feminine Products",
            "Phone Charger (iphone)",
            "Phone Charger (android)",
            "add item",
          ]
        }
        key="7"
        neededItems={Array []}
        setItems={[MockFunction]}
        setNeededItems={[MockFunction]}
      >
        <article
          className="Item"
        >
          <div
            id="add-item-container"
          >
            <input
              id="new-item"
              name="name"
              onChange={[Function]}
              placeholder="Add Item/Medication"
              type="text"
              value=""
            />
            <button
              id="submit-new-item"
              onClick={[Function]}
              type="submit"
            >
              <img
                alt="plus symbol"
                id="plus-img"
                src={
                  Object {
                    "ReactComponent": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                    "__esModule": true,
                    "default": "plus-sign.svg",
                  }
                }
              />
            </button>
          </div>
        </article>
      </Item>
    </article>
  </section>
</NeedsForm>
`;
